### ISSUE TITLE
Pre-push hooks seem to be ignored

### ISSUE THREAD

**Issue by nateridderman** [extrawurst/gitui#1933]
**Describe the bug**
When I push changes via gitui, it does not seem to run my repo's pre-push hook

My pre-push hook looks something like:
`#!/usr/bin/env bash`
`./gradlew detekt || exit $?`

**To Reproduce**
To stage/commit/push some changes via gitui with the aforementioned hook.

**Expected behavior**
When I run "git push" from the command line, my pre-push script is run (and stdout is shown in the terminal) and if it has an error, the push is aborted. For gitui, I would not expect it to show stdout, but the push dialog could have a state indicating that it's running pre-push hooks and waiting for the result.

**Screenshots**

**Context (please complete the following information):**
 - macOS 13.4.1
 - GitUI Version 0.24.3
 - Rust version: 1.72.1

**Comment by extrawurst** [comment]
yeah `pre-push` is currently not implemented: https://git-scm.com/docs/githooks#_pre_push

**Comment by argenkiwi** [comment]
After upgrading today whenever I try to push I get:
```
pre-push hook failed:
This should be run through Git's pre-push hook.  Run `git lfs update` to install it.
```
Unfortunately, running `git lfs update` does not fix the issue.

**Comment by extrawurst** [comment]
@argenkiwi please open a new issue for this and provide the relevant debug info a bug report requires. Please also provide infos like hooks you have configured and whether your repo uses lfs

**Comment by argenkiwi** [comment]
Sure, I have no time for it right now so I thought I would quickly bring this to your attention. The repo does use LFS, I'll have to do some research before I begin to understand what this change is about and why it is affecting me. Thanks.

**Comment by extrawurst** [comment]
followup as #2809
